src/System.Net.Http/src/System/Net/Http/Unix/CurlHandler.cs

```diff
@@ -93,6 +155,7 @@ protected override void Dispose(bool disposing)
             state.CancellationToken = cancellationToken;
             state.RequestMessage = request;
             state.Handler = this;
+            state.ProxyPolicy = _proxyPolicy;
```

Comment:
I am following the style/convention of winhttphandler. It makes it explicit which value we need to use for determining proxy behavior.